azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Route.java

```diff
@@ -0,0 +1,243 @@
+             * @param nextHopType a hop type
+             * @return the next stage of the update
+             */
+            Update withNextHop(RouteNextHopType nextHopType);
+        }
+    }
+
+    /**
+     * The entirety of a route update as part of a route table update.
+     */
+    interface Update extends
+        Settable<RouteTable.Update>,
+        UpdateStages.WithDestinationAddressPrefix,
+        UpdateStages.WithNextHopToVirtualAppliance,
+        UpdateStages.WithNextHopType {
+    }
+
+    /**
+     * Grouping of route definition stages applicable as part of a route table update.
+     */
+    interface UpdateDefinitionStages {
+        /**
+         * The first stage of a route definition.
+         * @param <ParentT> the return type of the final {@link WithAttach#attach()}
+         */
+        interface Blank<ParentT> extends WithDestinationAddressPrefix<ParentT> {
+        }
+
+        /** The final stage of a route definition.
+         * <p>
+         * At this stage, any remaining optional settings can be specified, or the route definition
+         * can be attached to the parent route table definition using {@link WithAttach#attach()}.
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithAttach<ParentT> extends
+            Attachable.InUpdate<ParentT> {
+        }
+
+        /**
+         * The stage of a route definition allowing to specify the destination address prefix.
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithDestinationAddressPrefix<ParentT> {
+            /**
+             * Specifies the destination address prefix to apply the route to.
+             * @param cidr an address prefix expressed in the CIDR notation
+             * @return the next stage of the definition
+             */
+            WithNextHopType<ParentT> withDestinationAddressPrefix(String cidr);
+        }
+
+        /**
+         * The stage of a route definition allowing to specify the IP address of a virtual appliance
+         * to direct the traffic through.
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithNextHopToVirtualAppliance<ParentT> {
+            /**
+             * Specifies the IP address of the virtual appliance for the next hop to go to.
+             * @param ipAddress an IP address of an existing virtual appliance (virtual machine)
+             * @return the next stage of the definition
+             */
+            WithAttach<ParentT> withNextHopToVirtualAppliance(String ipAddress);
+        }
+
+        /**
+         * The stage of a route definition allowing to specify the next hop type.
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithNextHopType<ParentT> extends WithNextHopToVirtualAppliance<ParentT> {
```

Comment:
good catch, it's a leftover of another idea i was exploring but wound up doing this in a simpler way. Addressed in an upcoming commit.